includes/Core/Modules/Module.php

```diff
@@ -411,18 +426,7 @@ protected function get_datapoint_services() {
 	 * @return array Map of datapoints to their definitions.
 	 */
 	protected function get_datapoint_definitions() {
-		$services = $this->get_datapoint_services();
-
-		return array_reduce(
-			array_keys( $services ),
-			function ( $map, $datapoint ) use ( $services ) {
-				$map[ "GET:$datapoint" ]  = array( 'service' => $services[ $datapoint ] );
-				$map[ "POST:$datapoint" ] = array( 'service' => $services[ $datapoint ] );
-
-				return $map;
-			},
-			array()
-		);
+		return array();
```

Comment:
Technically, I think this code should be maintained for backward-compatibility. But since it's unlikely anybody other than us is using this, I'm definitely up for discussing this. _If_ we get rid of the BC layer, then we should properly implement it all the way though, i.e. make `get_datapoint_definitions` abstract, because that's what it would have been if we had originally built it without BC consideration.

Reply:
In my opinion, this method doesn't have to be abstract no matter we support BC or not. The purpose of this function is deterministic, the function is expected to have no side effects and not to be affected outside changes, plus no matter what happens it should return an array in all children implementations. Thus we can extract its default beheviour (which is returning an array) and put it in the parent class - exactly what we have here.

Reply:
That's fair, there could be modules that don't define any datapoints anyway.